### DISCUSSION TITLE
Quarkus code format - Any article about?

### DISCUSSION THREAD

**Discussion by mhagnumdw** [quarkusio/quarkus#40542] [Community]
I found it interesting how the Quarkus team treats code formatting in Quarkus. Using maven and/or eclipse. How do you force the project's code style. I saw references to the plugins `net.revelc.code.formatter:formatter-maven-plugin` and `com.diffplug.spotless:spotless-maven-plugin`.

Is there any article of yours about this? For those who want to delve a little deeper.

Tks!

**Comment by maxandersen** [comment]
We don't have a dedicated article on it besides contributing: https://github.com/quarkusio/quarkus/blob/7f8a039533d88d76977b9503722e9a2167e112c2/CONTRIBUTING.md?plain=1#L256

but yes, those plugins is what is used and what we do is we have the `ide-config` artifact that all the projects uses.

The source is here: https://github.com/quarkusio/quarkus/tree/main/independent-projects/ide-config

Let me know if that covers it for you?

**Comment by gsmet** [comment]
That's actually a nice blog post idea. I'll put it on my list.
